Bottom: 143c699a373dc8318cadc73f45fe932ee2c97b48
Top:    9b96978a1d2d76e9921abaf5be38b2df3d781601
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-05-08 12:05:50 +0100

Refresh of workaround-crash-freeing

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 9df24ee..f73fb6b 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -3654,9 +3654,22 @@ poly_DelContour (PLINE ** c)
 
       g_slice_free (VNODE, cur);
     }
-  if ((*c)->head.cvc_next != NULL)
+
+  if ((*c)->head.cvc_next == -1)
+    {
+      fprintf (stderr, "WHY DID THIS HAPPEN?? cvc_next == -1 in poly_DelContour() !\n");
+    }
+  else if ((*c)->head.cvc_next != NULL)
     {
       free ((*c)->head.cvc_next);
+    }
+
+  if ((*c)->head.cvc_prev == -1)
+    {
+      fprintf (stderr, "WHY DID THIS HAPPEN?? cvc_prev == -1 in poly_DelContour() !\n");
+    }
+  else if ((*c)->head.cvc_prev != NULL)
+    {
       free ((*c)->head.cvc_prev);
     }
   /* FIXME -- strict aliasing violation.  */
